kolibri/plugins/learn/assets/src/state/actions/main.js

```diff
@@ -177,10 +177,14 @@ function showChannels(store) {
         return;
       }
       const channelRootIds = channels.map(channel => channel.root);
-      ContentNodeResource.getCollection({ ids: channelRootIds }).fetch().then(rootNodes => {
-        const pageState = {
-          rootNodes: _collectionState(rootNodes),
-        };
+      ContentNodeResource.getCollection({ ids: channelRootIds }).fetch().then(channelCollection => {
+        const rootNodes = _collectionState(channelCollection);
+        rootNodes.forEach(rootNode => {
+          rootNode.thumbnail = channels.find(
+            channel => channel.id === rootNode.channel_id
+          ).thumbnail;
```

Comment:
Not sure if this could happen in practice, but this will give you an error if `channels.find(...)` returns `undefined.

Reply:
If it does, something very odd has happened (we have a channel without a root node), may be worth catching just in case, but should be highly unlikely.